File: Data Structures & Algorithms/products-of-array-discluding-self/submission-4.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        result = [0] * len(nums)
        prefix = [0] * len(nums)
        postfix = [0] * len(nums)

        prefix[0] = nums[0]
        postfix[-1] = nums[-1]

        j = len(nums) - 2
        for i in range(1, len(nums), 1):
            prefix[i] = prefix[i - 1] * nums[i]
            postfix[j] = nums[j] * postfix[j + 1]
            j -= 1
        
        for i in range(len(result)):
            if i == 0:
                result[i] = 1 * postfix[i + 1]
            elif i == len(result) - 1:
                result[i] = prefix[i - 1] * 1
            else:
                result[i] = prefix[i - 1] * postfix[i + 1]

        return result

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int temp = 0;
        int result = 1;

        TreeSet<Integer> sortedSet = new TreeSet<>();
        PriorityQueue<Integer> maxHeap = new PriorityQueue<>(Collections.reverseOrder());

        if(nums.length > 0)
        {
            for(int num : nums) 
            {
                sortedSet.add(num);
            }

            temp = sortedSet.pollFirst();

            for(Integer element : sortedSet)
            {
                if(temp + 1 == element)
                {
                    result += 1;
                }
                else
                {
                    maxHeap.add(result);
                    result = 1;
                }

                temp = element;
            }

            maxHeap.add(result);

            result = maxHeap.peek();
        }
        else
        {
            result = 0;
        }

        return result;
    }
}
